src/crepe/api/Button.h

```diff
@@ -15,43 +27,22 @@ class Button : public UIObject {
 	 * \param id The unique ID of the game object associated with this button.
 	 * \param dimensions The width and height of the UIObject
 	 * \param offset The offset relative this GameObjects Transform
-	 * \param on_click callback function that will be invoked when the button is clicked.
 	 */
-	Button(game_object_id_t id, const vec2 & dimensions, const vec2 & offset,
-		   const std::function<void()> & on_click);
-
-	// TODO: create separate toggle button class
-	/**
-	 * \brief The callback function to be executed when the button is clicked.
-	 *
-	 * This function is invoked whenever the button is clicked. It can be set to any
-	 * function that matches the signature `void()`.
-	 */
-	std::function<void()> on_click = nullptr;
-
+	Button(game_object_id_t id, const vec2 & dimensions, const vec2 & offset);
 	/**
-	 * \brief Callback function to be executed when the mouse enters the button's boundaries.
+	 * \brief Get the maximum number of instances for this component
 	 *
-	 * This function is triggered when the mouse cursor moves over the button, allowing
-	 * custom actions like visual effects, highlighting, or sound effects.
+	 * Since the button Event transfers the GameObject Metadata it will be the same for each button so only one button is allowed per GameObject	
+	 * 
+	 * \return The maximum number of instances for this component
```

Comment:
Hier mag je ook direct `\returns 1` neerzetten omdat deze functie als variabele fungeert en de waarde dan direct in de documentatie zichtbaar wordt.